### PR TITLE
Refactor Apple Wallet code (#76)

### PR DESCRIPTION
**Pull request by aahna-ashina**
## Dework Task

https://app.dework.xyz/nation3/passport-services-p?taskId=ae6efa25-b222-4700-afc8-63e65b870cb9

## Related GitHub Issue

#76
   - [x] https://github.com/nation3/mobile-passport/pull/91/commits/df21a4d1d418e4d3e608024ab65e22a3307a55a1 Add a simple helper for the to/from milliseconds calculations
   - [x] https://github.com/nation3/mobile-passport/pull/91/commits/ecf3d0c571a2528fc05e56741c5b654fa88511aa Reduce the number of branches of code in `[passTypeIdentifier].ts` (https://github.com/nation3/mobile-passport/pull/79#discussion_r972628309)
   - [x] https://github.com/nation3/mobile-passport/pull/91/commits/bbfa5916bca261c382b81927890d06c50fe44098 Separate APN provider code in `Passes.ts` into a standalone file/function (https://github.com/nation3/mobile-passport/pull/79#discussion_r972628309)

Improve stability of APNs integration:
- [x] https://github.com/nation3/mobile-passport/pull/91/commits/36f5e8d56ce517b5beb892d77a8fe5b0e0aad2d5 Improve asynchronous calls to APNs in `Passes.ts`, so that a JSON response is not returned until all notification requests have been confirmed by APNs.

## How Has This Been Tested?

- [x] Status checks pass
- [x] https://github.com/nation3/mobile-passport/pull/91/commits/138492a0cbf9225c29a681489c62e3bdcd8fc7b5 Mock APN provider during unit tests 
- [x] Increased code coverage from 80% to 85%
- [x] Works on Goerli
- [ ] Works on Mainnet

## Are There Admin Tasks?

Nada

### CODE REVIEW

**Comment by aahna-ashina** [comment]
@luisivan This PR improves calls to `/api/pushLastUpdate` by returning a more details JSON response:
```json
{
    "summary": "1 sent, 3 failed",
    "sent": [ ... ],
    "failed": [ ... ]
}
```

Also the code is changed from async to `await`, so that a response is not returned until _all_ push notification requests have been sent.  For now, the waiting time is just a few seconds.  And in the future, with hundreds/thousands of registered passes, we should improve the user experience by building a UI where you can see a progress bar while sending out pass updates.

File: server/jest.config.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ module.exports = {
   testEnvironment: 'node',
   coverageThreshold: {
     global: {
-      lines: 80.00
+      lines: 79.00
```
Comment (johnmark13) [inline server/jest.config.js:7]: 😄

**Review by johnmark13 — APPROVED** [review]

File: server/jest.config.js (hunk 1 above)
Comment (aahna-ashina) [reply server/jest.config.js:7]: How sad 🥲

**Comment by aahna-ashina** [comment]
@luisivan @okhaimie-dev @xPi2 Need your review before merging 🙏

**Review by okhaimie-dev — APPROVED** [review]

**Comment by aahna-ashina** [comment]
@johnmark13 Added mocking:  https://github.com/nation3/mobile-passport/pull/91/commits/138492a0cbf9225c29a681489c62e3bdcd8fc7b5

**Review by xPi2 — APPROVED** [review]
LGTM, but maybe I'm missing enough context to give better feedback.

File: server/jest.config.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ module.exports = {
   testEnvironment: 'node',
   coverageThreshold: {
     global: {
-      lines: 80.00
+      lines: 85.00
```
Comment (johnmark13) [inline server/jest.config.js:7]: 💪

**Review by johnmark13 — APPROVED** [review]
I think this is really clever 💪

**Pull request merged**